test/bash/util.bats

```diff
@@ -0,0 +1,158 @@
+
+setup() {
+  load '../../node_modules/bats-support/load'
+  load '../../node_modules/bats-assert/load'
+
+  TEST_DIR="$( cd "$( dirname "$BATS_TEST_FILENAME" )" >/dev/null 2>&1 && pwd )"
+  TEST_ROOT_DIR=$(mktemp -u)
+
+  # Not used yet, but will be later
+  load "$TEST_DIR/../../src/usr/local/buildpack/util.sh"
+}
+
+teardown() {
+  rm -rf "${TEST_ROOT_DIR}"
+}
+
+@test "require_tool" {
+  run require_tool
+  assert_output --partial "No tool defined"
+  assert_failure
+
+  # TODO(Chumper): In my opinion this test should fail
+  run require_tool foo
+  # assert_output --partial "No version defined"
+  # assert_failure
+  assert_success
+
+  require_tool foo 1.2.3
+  assert [ "${TOOL_NAME}" = "foo" ]
+  assert [ "${TOOL_VERSION}" = "1.2.3" ]
+  assert [ "${FOO_VERSION}" = "1.2.3" ]
+
+  FOO_VERSION=1.2.4 require_tool foo
+  assert [ "${TOOL_NAME}" = "foo" ]
+  assert [ "${TOOL_VERSION}" = "1.2.4" ]
+
+  FOO_BAR_VERSION=1.2.5 require_tool foo-bar
+  assert [ "${TOOL_NAME}" = "foo-bar" ]
+  assert [ "${TOOL_VERSION}" = "1.2.5" ]
+}
+
+@test "requires_user" {
+  run require_user
+  assert_output --partial "No USER_NAME defined"
+  assert_failure
+
+  # TODO(Chumper): In my opinion this test should fail
+  USER_NAME= run require_user
+  # assert_output --partial "No USER_NAME defined"
+  # assert_failure
+  assert_success
```

Comment:
@viceice Same here, I would like your opinion on this. 
Is this working as intended?

Reply:
should fail